tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -297,6 +297,7 @@ def parse_datafile_lines(datafile, dfs, program_type, section, is_encrypted, cas
                 case_consistency_validator.add_record(record, s, len(record_errors) > 0)
 
         # Add any generated cat4 errors to our error data structure & clear our caches errors list
+        num_errors += case_consistency_validator.num_generated_errors()
         unsaved_parser_errors[None] = unsaved_parser_errors.get(None, []) + \
             case_consistency_validator.get_generated_errors()
```

Comment:
With respect to our async chat, this is the exact spot that the change you made in [this](https://github.com/raft-tech/TANF-app/pull/2859/commits/1e9cf8481a2c0df6a1ca817b58ecdc53911b53cc) could have a negative side effect. There is a potential for duplicate errors to be created here. I think it won't be an issue since we don't generate any errors until we have validated. But perhaps you could do some checking/check my logic here to get a warm and fuzzy?

Reply:
i agree there's a potential for duplicates, that's what worried me initially. we get away with it by clearing `generated_errors` after each call to `validate` in `parse_datafile_lines`. as long as the list is cleared, this should work